astropy/utils/masked/core.py

```diff
@@ -995,10 +995,80 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
                 # We don't support this.
                 return NotImplemented
 
-        elif method in {"reduceat", "at"}:  # pragma: no cover
-            raise NotImplementedError(
-                "masked instances cannot yet deal with 'reduceat' or 'at'."
-            )
+        elif method == "reduceat":
+            # Combine input mask with where_mask
+            combined_mask = combine_masks((masks[0], where_mask), copy=False)
```

Comment:
Note that ufunc.reduceat does not take a `where` argument (see https://numpy.org/doc/stable/reference/generated/numpy.ufunc.reduceat.html), so `where_mask` is guaranteed not to be set. In principle, however, `indices` could be masked -- I think it would be fine to just fail on that, but we should check and test that path.